GERTi/GERTiClient.lua

```diff
@@ -26,23 +26,83 @@ if (component.isAvailable("tunnel")) then
 end
 
 -- addresses
-local eAddress = 0
-local iAddress = 0
+
+-- This only matters to the "GERTi" API way down below. Should be int - see startup procedure.
+-- It is updated by handler["RegisterComplete"] if the message concerns this node.
+local iAddress = nil
 local cachedAddress = {{}, {}, {}, {}, {}}
 local addressDex = 1
 -- Tables of neighbors and connections
+-- [x]{
+--  address
+--  port
+--  tier
+-- }
 local neighbors = {}
 local tier = 3
 local neighborDex = 1
+-- [x]{
+--  destination
+--  origin
+--  beforeHop
+--  nextHop
+--  port
+--  data
+--  dataDex
+-- }
 local connections = {}
 local connectDex = 1
+
+-- Addresses awaiting resolution. Key is the gAddress, value is the function.
+-- This is as opposed to addTempHandler which has no caching.
+-- Add to this with addWaitingAddress.
+-- These are cancelled after a set time.
+local waitingAddress = {}
+
 local handler = {}
 
 -- internal functions
 local function sortTable(elementOne, elementTwo)
 	return (tonumber(elementOne["tier"]) < tonumber(elementTwo["tier"]))
 end
 
+-- like sortTable : these are common between MNC & GERTiClient
+-- this function adds a handler for a set time in seconds, or until that handler returns a truthful value (whichever comes first)
+-- extremely useful for callback programming
+local function addTempHandler(timeout, code, cb, cbf)
+	local function cbi(...)
+		local evn, rc, sd, pt, dt, code2 = ...
+		if code ~= code2 then return end
+		if cb(...) then
+			-- Returning false from the event handler (specifically false) will get rid of the handler.
+			-- I have no idea why an event handler in this code was set up to cause it to return false,
+			--  apart from this one (because it wasn't *the handler holding everything together*)
+			return false
+		end
+	end
+	event.listen("modem_message", cbi)
+	event.timer(timeout, function ()
+		event.ignore(cbi)
+		cbf()
+	end)
+end
+-- Like a sleep, but it will exit early if a modem_message is received and then something happens.
+local function waitWithCancel(timeout, cancelCheck)
```

Comment:
Event.pull already does this, so this entire function appears unnecessary.

Reply:
It does not allow cancelling the wait *if a specific condition triggers*, which is crucial.

Reply:
Will review again after other requested changes are made